### DISCUSSION TITLE
Using f90charm, how do I retrieve 2-D array data from a chare when it's done doing work?

### DISCUSSION THREAD

**Discussion by fclingen** [UIUC-PPL/charm#3375] [Q&A] [answered]
I have a legacy Fortran code I'm trying to parallelize.  Essentially, it's a loop that modfies a lot of 2-D arrays in-place, and I want to parallelize the loop to pass those arrays into chare, do the modifications, and replace the data in the main thread with that from the workers.

I understand how to instantiate the chares with the Fortran90 bindings and get them cranking on the calcs, but I'm unable to find anything in the f90charm documentation that tells me how CkCallbacks work in Fortran to get my results back when the work is finished.  How do I get my data back to pe0?

**Comment by evan-charmworks** [comment, marked as answer]
You'll need to perform a reduction. Have a look at `examples/charm++/f90charm/ldbRedDemo`.

**Reply by fclingen-pet** [reply]
LdbRedDemo is performing a sum reduction on a single array accross chares, yes?  I'm not sure how that can be coaxed into sending the data into a single pe?  Call contribute with CHARM_SET instead of CHARM_SUM_INT and point into the resultant array?
